File: bench/01_dollar_mm.R
library(torch)

batch_size <- as.numeric(Sys.getenv("BATCH_SIZE", unset = "1000"))
device <- Sys.getenv("DEVICE", unset = "cpu")

w <- torch_randn(784, 512, device=device)
x <- torch_randn(batch_size, 784, device=device)

f <- function() {
  for (i in 1:iter) {
    x$mm(w)
  }
  invisible(NULL)
}

if (device == "cpu") {
  fn <- f
} else {
  fn <- function() {
    f()
    torch:::cuda_synchronize()
  }
}

iter <- 1
fn()
iter <- as.numeric(Sys.getenv("ITER", unset = "1000"))

cat(system.time(fn())[["elapsed"]])

File: bench/01_dollar_mm.py
import torch
import time
import os

batch_size = int(os.environ.get('BATCH_SIZE', "1000"))
device = os.environ.get('DEVICE', "cpu")

x = torch.randn(batch_size, 784, device=device)
w = torch.randn(784, 512, device=device)

def f ():
  for i in range(ir):
    x.mm(w)
  
  return None

if device=="cpu":
  fn = f
else:
  def fn():
    f()
    torch.cuda.synchronize()

ir = 1
fn()
ir = int(os.environ.get('ITER', "1000"))

start_time = time.time()
fn()
print((time.time() - start_time))




